maven/org.eclipse.oomph.maven.generator/report/m2e/updated.target

```diff
@@ -121,5 +121,49 @@
 				</dependency>
 			</dependencies>
 		</location>
+		<location includeDependencyDepth="none" includeDependencyScopes="compile" includeSource="true" label="maven-archetype" missingManifest="generate" type="Maven">
+			<dependencies>
+				<dependency>
+					<groupId>org.apache.maven.archetype</groupId>
+					<artifactId>archetype-catalog</artifactId>
+					<version>3.2.1</version>
+					<type>jar</type>
+				</dependency>
+				<dependency>
+					<groupId>org.apache.maven.archetype</groupId>
+					<artifactId>archetype-common</artifactId>
+					<version>3.2.1</version>
+					<type>jar</type>
+				</dependency>
+				<dependency>
+					<groupId>org.apache.maven.archetype</groupId>
+					<artifactId>archetype-descriptor</artifactId>
+					<version>3.2.1</version>
+					<type>jar</type>
+				</dependency>
+				<dependency>
+					<groupId>org.apache.maven.shared</groupId>
+					<artifactId>maven-artifact-transfer</artifactId>
+					<version>0.13.1</version>
+					<type>jar</type>
+				</dependency>
+				<dependency>
+					<groupId>org.jdom</groupId>
+					<artifactId>jdom2</artifactId>
+					<version>2.0.6.1</version>
+					<type>jar</type>
+				</dependency>
+			</dependencies>
+			<instructions><![CDATA[
+Bundle-Name:           M2Eclipse supplement fragment for ${mvnGroupId}:${mvnArtifactId}:${mvnVersion}
+version:               ${version_cleanup;${mvnVersion}}
+Bundle-SymbolicName:   org.eclipse.m2e.archetype.${mvnArtifactId}
+Bundle-Version:        ${version}
+Export-Package:        *;version="${version}";-noimport:=true
+Fragment-Host:         org.eclipse.m2e.archetype.common;bundle-version="[3,4)"
+-exportcontents:       META-INF.plexus;-noimport:=true;x-internal:=true,org.apache.maven.archetype.*;provider=m2e;mandatory:=provider;version="${version}";x-friends:="org.eclipse.m2e.core.ui"
+Import-Package:        !*
+]]></instructions>
+		</location>
```

Comment:
These maven archetype artifacts are not fully functional and are stripped down to whats necessary to make them work only for the use case of M2E. They are tightly coupled to m2e and are not general purpose wrappings.
Therefore I think they should be included here.

Reply:
This is where I've been doing the BND wrapping:

https://github.com/eclipse-oomph/oomph.incubator/blob/master/maven-bnd/tp/MavenBND.target

Now that I'm one GitHub, I'm going to create proper read-mes to document everything better...

Of course this particular 'maven-bnd' is probably something people want moved to Orbit and I still need to discuss with Jonah if this m2e location approach is the best way.   I tried to added poi this weekend and thrashed for the better part of a day chasing down dependencies.  😭

Reply:
> Of course this particular 'maven-bnd' is probably something people want moved to Orbit and I still need to discuss with Jonah if this m2e location approach is the best way.

Yes that would make sense to have that in Orbit or its (future) derivatives. Personally I'm not sure either and it probably depends on the complexity of the wrapping, e.g. if only the Manifest is generated or if the jars are even re-packed.

In case you want another participant, I would be interested about that discussion as well.